From: Peter Volkov <pva@gentoo.org>
Date: Sun, 30 Nov 2008 13:48:39 +0000 (+0300)
Subject: Added unlimited CT config
X-Git-Url: http://git.openvz.org/?p=vzctl;a=commitdiff_plain;h=c8dc6e5a60137eafca882c39ffbeb0177e65a064

Added unlimited CT config

This implements (again thank Daniel Robbins for suggestion and initial
work) ve-unlimited.conf-sample configuration file. It sets CT resources
to unlimited values. Not to repeating his reasoning I cite him here:

"I think that adding this to the /etc/vz/conf directory would be a big
step forward. I think that OpenVZ is going to be used more and more for
single-developer workstations where developers want to use containers to
create test environments. In these single-user situations, resource limits
just get in the way."

I tend to aggree. If you just want to use containers like very smart
chroots resource limits just make things harder.
---

diff --git a/etc/conf/Makefile.am b/etc/conf/Makefile.am
index 963abdc..52c16e8 100644
--- a/etc/conf/Makefile.am
+++ b/etc/conf/Makefile.am
@@ -18,7 +18,7 @@
 include $(top_srcdir)/paths.am
 
 VE0CONFIG = 0.conf
-VECONFIG  = ve-light.conf-sample ve-vps.basic.conf-sample
+VECONFIG  = ve-light.conf-sample ve-vps.basic.conf-sample ve-unlimited.conf-sample
 
 if ARCH_IA64
 VECONFIG_EXT = .ia64
diff --git a/etc/conf/ve-unlimited.conf-sample b/etc/conf/ve-unlimited.conf-sample
new file mode 100644
index 0000000..2b9febd
--- /dev/null
+++ b/etc/conf/ve-unlimited.conf-sample
@@ -0,0 +1,51 @@
+#  Copyright (C) 2000-2008, Parallels, Inc. All rights reserved.
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+
+VERSION="2"
+
+ONBOOT="no"
+
+# UBC parameters (in form of barrier:limit)
+# Primary parameters
+AVNUMPROC="unlimited"
+NUMPROC="unlimited"
+NUMTCPSOCK="419430"
+NUMOTHERSOCK="419430"
+VMGUARPAGES="unlimited"
+# Secondary parameters
+KMEMSIZE="unlimited"
+#TCPSNDBUF="2147483647:2147483647"
+TCPSNDBUF="1073741823:2147483647"
+TCPRCVBUF="1073741823:2147483647"
+OTHERSOCKBUF="1073741823:2147483647"
+DGRAMRCVBUF="unlimited"
+OOMGUARPAGES="unlimited"
+# Auxiliary parameters
+LOCKEDPAGES="unlimited"
+SHMPAGES="unlimited"
+PRIVVMPAGES="unlimited"
+NUMFILE="unlimited"
+NUMFLOCK="unlimited"
+NUMPTY="unlimited"
+NUMSIGINFO="unlimited"
+DCACHESIZE="unlimited"
+
+PHYSPAGES="0:unlimited"
+NUMIPTENT="unlimited"
+
+# Disk quota parameters (in form of softlimit:hardlimit)
+DISK_QUOTA="no"
diff --git a/etc/conf/ve-unlimited.conf-sample.ia64 b/etc/conf/ve-unlimited.conf-sample.ia64
new file mode 100644
index 0000000..42fd7f6
--- /dev/null
+++ b/etc/conf/ve-unlimited.conf-sample.ia64
@@ -0,0 +1,51 @@
+#  Copyright (C) 2000-2008, Parallels, Inc. All rights reserved.
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+
+VERSION="2"
+
+ONBOOT="no"
+
+# UBC parameters (in form of barrier:limit)
+# Primary parameters
+AVNUMPROC="unlimited"
+NUMPROC="unlimited"
+NUMTCPSOCK="1801439850948198"
+NUMOTHERSOCK="1801439850948198"
+VMGUARPAGES="unlimited"
+# Secondary parameters
+KMEMSIZE="unlimited"
+#TCPSNDBUF="2147483647:2147483647"
+TCPSNDBUF="4611686018427387903:9223372036854775807"
+TCPRCVBUF="4611686018427387903:9223372036854775807"
+OTHERSOCKBUF="4611686018427387903:9223372036854775807"
+DGRAMRCVBUF="unlimited"
+OOMGUARPAGES="unlimited"
+# Auxiliary parameters
+LOCKEDPAGES="unlimited"
+SHMPAGES="unlimited"
+PRIVVMPAGES="unlimited"
+NUMFILE="unlimited"
+NUMFLOCK="unlimited"
+NUMPTY="unlimited"
+NUMSIGINFO="unlimited"
+DCACHESIZE="unlimited"
+
+PHYSPAGES="0:unlimited"
+NUMIPTENT="unlimited"
+
+# Disk quota parameters (in form of softlimit:hardlimit)
+DISK_QUOTA="no"
diff --git a/etc/conf/ve-unlimited.conf-sample.x86_64 b/etc/conf/ve-unlimited.conf-sample.x86_64
new file mode 100644
index 0000000..42fd7f6
--- /dev/null
+++ b/etc/conf/ve-unlimited.conf-sample.x86_64
@@ -0,0 +1,51 @@
+#  Copyright (C) 2000-2008, Parallels, Inc. All rights reserved.
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+
+VERSION="2"
+
+ONBOOT="no"
+
+# UBC parameters (in form of barrier:limit)
+# Primary parameters
+AVNUMPROC="unlimited"
+NUMPROC="unlimited"
+NUMTCPSOCK="1801439850948198"
+NUMOTHERSOCK="1801439850948198"
+VMGUARPAGES="unlimited"
+# Secondary parameters
+KMEMSIZE="unlimited"
+#TCPSNDBUF="2147483647:2147483647"
+TCPSNDBUF="4611686018427387903:9223372036854775807"
+TCPRCVBUF="4611686018427387903:9223372036854775807"
+OTHERSOCKBUF="4611686018427387903:9223372036854775807"
+DGRAMRCVBUF="unlimited"
+OOMGUARPAGES="unlimited"
+# Auxiliary parameters
+LOCKEDPAGES="unlimited"
+SHMPAGES="unlimited"
+PRIVVMPAGES="unlimited"
+NUMFILE="unlimited"
+NUMFLOCK="unlimited"
+NUMPTY="unlimited"
+NUMSIGINFO="unlimited"
+DCACHESIZE="unlimited"
+
+PHYSPAGES="0:unlimited"
+NUMIPTENT="unlimited"
+
+# Disk quota parameters (in form of softlimit:hardlimit)
+DISK_QUOTA="no"
